evalml/tests/automl_tests/test_engine_base.py

```diff
@@ -127,26 +117,25 @@ def test_train_pipeline_trains_and_tunes_threshold(
     automl_config = AutoMLConfig(
         None, "binary", LogLossBinary(), [], None, True, None, 0, None, None
     )
-    _ = train_pipeline(
-        dummy_binary_pipeline_class({}), X, y, automl_config=automl_config
-    )
+    env = AutoMLTestEnv("binary")
+    with env.test_context():
+        _ = train_pipeline(
+            dummy_binary_pipeline_class({}), X, y, automl_config=automl_config
+        )
 
-    mock_pipeline_fit.assert_called_once()
-    mock_optimize.assert_not_called()
+    env.mock_fit.assert_called_once()
+    env.mock_optimize_threshold.assert_not_called()
```

Comment:
This is a great observation. The mocks are not set as class attributes until after the `yield` in the context manager, i.e until after the computation we're mocking has finished running. So the mocks have to be accessed outside `test_manager`!

I modified the implementation so that a `ValueError` is raised if mock is accessed within `test_context`. Hopefully that'll help any future confusion.